### DISCUSSION TITLE
Solution for keymap to copy branch name

### DISCUSSION THREAD

**Discussion by khaitranhq** [NeogitOrg/neogit#1520] [Q&A] [answered]
Hi everyone.
My job requires frequent copying of the current branch name, but I can't locate a key map that supports this action. I also searched through the API, but I didn't find any useful information.
The only option I have is to edit the branch description and copy the branch name, which requires a lot of keystrokes.
Thank you for any support you can give.

**Comment by CKolkey** [comment, marked as answer]
You could bind a keymapping to this function to accomplish that:

```lua
function()
  local current_branch = require("neogit.lib.git").branch.current()
  vim.cmd.let(("@+='%s'"):format(current_branch))
end
```

Hope that helps :)

**Comment by khaitranhq** [comment]
thank you! it worked
